Sources/HummingbirdCore/Response/ResponseBodyWriter.swift

```diff
@@ -15,67 +15,24 @@
 import HTTPTypes
 import NIOCore
 
-/// HTTP Response Body part writer
-public protocol ResponseBodyWriter {
-    /// Write a single ByteBuffer
-    /// - Parameter buffer: single buffer to write
-    mutating func write(_ buffer: ByteBuffer) async throws
-    /// Write a sequence of ByteBuffers
-    /// - Parameter buffers: Sequence of buffers
-    mutating func write(contentsOf buffers: some Sequence<ByteBuffer>) async throws
-    /// Finish writing body
-    /// - Parameter trailingHeaders: Any trailing headers you want to include at end
-    consuming func finish(_ trailingHeaders: HTTPFields?) async throws
-}
+/// Type erasing ``ResponseBodyWriterProtocol``.
+///
+/// Holds a pointer to any ResponseBodyWriterProtocol.
+public struct ResponseBodyWriter: ResponseBodyWriterProtocol {
+    public var wrapped: any ResponseBodyWriterProtocol
```

Comment:
I wonder if sometime in the future you'd want to make an special case for hummingbird's own response-body-writer with an enum, for performance optimizations.

Just something to keep in mind. This public var shouldn't stop use from doing that.

Reply:
(Like `NIOAny`, or even String's own "guts")